VocaDbWeb.Core/Controllers/StatsController.cs

```diff
@@ -0,0 +1,805 @@
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Net.Http.Headers;
+using VocaDb.Model.Database.Queries;
+using VocaDb.Model.Database.Repositories;
+using VocaDb.Model.DataContracts.Aggregate;
+using VocaDb.Model.Domain;
+using VocaDb.Model.Domain.Activityfeed;
+using VocaDb.Model.Domain.Albums;
+using VocaDb.Model.Domain.Artists;
+using VocaDb.Model.Domain.Globalization;
+using VocaDb.Model.Domain.PVs;
+using VocaDb.Model.Domain.Security;
+using VocaDb.Model.Domain.Songs;
+using VocaDb.Model.Domain.Tags;
+using VocaDb.Model.Domain.Users;
+using VocaDb.Model.Service.QueryableExtensions;
+using VocaDb.Model.Utils.Config;
+using VocaDb.Web.Code.Highcharts;
+using VocaDb.Web.Helpers;
+
+namespace VocaDb.Web.Controllers
+{
+	public class StatsController : ControllerBase
+	{
+		private const int ClientCacheDurationSec = 86400;
+		private readonly VdbConfigManager _config;
+		private readonly ActivityEntryQueries _activityEntryQueries;
+		private readonly StatsQueries _queries;
+		private readonly SongAggregateQueries _songAggregateQueries;
+
+		// TODO: implement
+		/*private T GetCachedReport<T>() where T : class
+		{
+			var name = ControllerContext.RouteData.Values["action"] + "_" + ControllerContext.RouteData.Values["cutoff"];
+			var item = _context.Cache["report_" + name];
+
+			if (item == null)
+				return null;
+
+			return (T)item;
+		}*/
+		private T GetCachedReport<T>() where T : class => throw new NotImplementedException();
+
+		// TODO: implement
+		/*private void SaveCachedReport<T>(T data) where T : class
+		{
+			var name = ControllerContext.RouteData.Values["action"];
+			_context.Cache.Add("report_" + name, data, null, Cache.NoAbsoluteExpiration, TimeSpan.FromDays(1), CacheItemPriority.Default, null);
+		}*/
+		private void SaveCachedReport<T>(T data) where T : class => throw new NotImplementedException();

```

Comment:
How can I implement these two methods? Should we use `ObjectCache`?

Reply:
Maybe just ResponseCache? But yes, the old ASP.NET cache can always be replaced with ObjectCache.